src/io/stapled.cr

```diff
@@ -0,0 +1,122 @@
+# This class staples together two unidirectional `IO`s to form a single,
+# bidirectional `IO`.
+#
+# Example (loopback):
```

Comment:
I would use another example, maybe using two `IO::Memory`. When I first saw this I thought "why not use `IO.pipe` directly?".